src/Lecture2.hs

```diff
@@ -81,7 +120,18 @@ lists of even lengths.
 ♫ NOTE: Use eta-reduction and function composition (the dot (.) operator)
   in this function.
 -}
-evenLists = error "TODO"
+{-ok, this is quite straightforward -}
+evenLists :: [[a]] -> [[a]]
+evenLists = removeLenghts . selectEven . computeLenghts
```

Comment:
You actually don't need to store lengths in the second element of a pair. You can use `filter` (as you already do) and calculate the length of the list each time.

After this, the function will become a short one-liner 🙂